### PR TITLE
Slack file upload plugin feature

### PR DESCRIPTION
**Pull request by senthilhns**
This code changes implement file upload on Slack using slack go

Following changes done
Bumped up github.com/slack-go/slack package to v0.15.0
Slack file upload plugin feature added
README documented for slack file upload usage
Unit test case for file upload added

Pipeline testing completed, pipeline id: slack_file_send_senthil02

https://app.harness.io/ng/account/gCoPSwHxS7ipOgx2iA9tOQ/all/orgs/default/projects/jenkinstoharness/pipelines/slack_file_send_senthil02/executions/xN4nrfnaQq-T3bmsQ2Hd5Q/pipeline

### CODE REVIEW

File: test/UnitTestDockerfile
Comment (Ompragash) [inline test/UnitTestDockerfile:1]: to be removed

File: plugin.go [hunk 1]
```diff
@@ -307,6 +319,133 @@ func (p Plugin) Exec() error {
...
+
+	return nil
+}
+
+func GetOutputVariablesStorageFilePath() string {
+	if os.Getenv("DRONE_OUTPUT") == "" {
+		return "/tmp/drone-output"
+	}
+	return os.Getenv("DRONE_OUTPUT")
+}
```
Comment (Ompragash) [inline plugin.go:439]:
```suggestion
```

we don't have to create a new file for DRONE_OUTPUT

File: plugin.go [hunk 2]
```diff
@@ -307,6 +319,133 @@ func (p Plugin) Exec() error {
...
+
+	valueStr := fmt.Sprintf("%v", value)
+
+	_, err = fmt.Fprintf(outputFile, "%s=%s\n", key, valueStr)
+	if err != nil {
+		return fmt.Errorf("failed to write to env: %w", err)
+	}
+
+	return nil
+}
```
Comment (Ompragash) [inline plugin.go:432]:
```suggestion
func WriteEnvToOutputFile(key, value string) error {
	outputFile, err := os.OpenFile(os.Getenv("DRONE_OUTPUT"), os.O_APPEND|os.O_CREATE|os.O_WRONLY, 0644)
	if err != nil {
		return fmt.Errorf("failed to open output file: %w", err)
	}

	defer outputFile.Close()

	_, err = fmt.Fprintf(outputFile, "%s=%s\n", key, value)
	if err != nil {
		return fmt.Errorf("failed to write to env: %w", err)
	}

	return nil
}
```

**Review by Ompragash — CHANGES_REQUESTED** [review]

File: test/UnitTestDockerfile
Comment (senthilhns) [reply test/UnitTestDockerfile:1]: done

File: plugin.go [hunk 3]
```diff
@@ -307,6 +318,117 @@ func (p Plugin) Exec() error {
...
+	}
+	return nil
+}
+
+func GetFileSize(filePath string) (int, error) {
+	fileInfo, err := os.Stat(filePath)
+	if err != nil {
+		return 0, err
+	}
+	return int(fileInfo.Size()), nil
```
Comment (Ompragash) [inline plugin.go:429]:
```suggestion
	fileInfo, err := os.Stat(filePath)
	if err != nil {
		return 0, fmt.Errorf("failed to get file info: %w", err)
	}
	if fileInfo.IsDir() {
		return 0, fmt.Errorf("path %s is a directory, not a file", filePath)
	}
	return int(fileInfo.Size()), nil
```

File: README.md [hunk 4]
```diff
@@ -60,6 +61,33 @@ Make sure to replace `your_access_token` with your actual Slack access token and
 
 If you provide an access token, it will use the Slack API to send the message. Otherwise, it will use the webhook.
 
+
+## Upload files on Slack Usage
```
Comment (Ompragash) [inline README.md:65]:
```suggestion
## Upload files to Slack
```

**Pull request merged**